Lib/idlelib/pyshell.py

```diff
@@ -1445,11 +1451,16 @@ def main():
         dir = os.getcwd()
         if dir not in sys.path:
             sys.path.insert(0, dir)
-    # check the IDLE settings configuration (but command line overrides)
+    # Check the IDLE settings configuration (but command line overrides).
     edit_start = idleConf.GetOption('main', 'General',
                                     'editor-on-startup', type='bool')
     enable_edit = enable_edit or edit_start
     enable_shell = enable_shell or not enable_edit
+    if idleConf.GetOption('main', 'ShellWindow',
+                          'startup-code-on', type='bool'):
+        config_startup_code = idleConf.GetOption(
+                'main', 'ShellWindow', 'shell-startup-code')
+        sys.argv = ['-c'] + [config_startup_code]
```

Comment:
When I use '-c' in Windows command prompt, the string that follows must be quoted with "", not '' (not an issue when provided in a variable) and *cannot contain literal newlines*.  I know that Linux shells are different.  I suspect that the newlines in the test strings might be the test failure cause.

Even if the test passes, I would not be sure that this line will work as expected.  User code must be executed by run.py in its pseudomain context after Python starts run.py in the real main context.

Reply:
It worked on Linux and I wasn't sure what issues it would run into on other systems.  Although I was considering the config code to be similar to the '-c' or -'s' options in terms of functionality.  But I didn't realize that those might not currently be correct.